graph_snyk_cve_sync.py

```diff
@@ -267,131 +295,20 @@ def _parse_maven_data(self, maven_data):
-        logger.info("Total vulnerabilities: {}".format(total_vuln))
-        logger.debug(self.CVE_DATA['maven'])
-
-    def _parse_pypi_data(self, pypi_data):
-        """Parse data for pypi."""
-        total_vuln = 0
-        pypi_pkg_versions = {}
-        if len(pypi_data) != 0:
-            self.CVE_DATA['pypi'] = {}
-            for data in pypi_data:
-                pkg = data['package']
-                logger.debug("Fetching details for package: {}".format(pkg))
-                try:
-                    versions = None
-                    if len(data['vulnerableVersions']) == 0:
-                        logger.info("False positive pypi {} being ignored.".format(pkg))
-                        continue
-                    if pkg not in self.CVE_DATA['pypi']:
-                        resp_obj = get_versions_and_latest_for_ep('pypi', pkg)
-                        if resp_obj and (not isinstance(resp_obj, list)):
-                            versions = resp_obj['versions']
-                            self.CVE_DATA['pypi'][pkg] = {
-                                "affected": [],
-                                "vulnerabilities": [],
-                                "all_ver": versions,
-                                "latest_version": resp_obj['latest_version'],
-                                "ecosystem": "pypi",
-                                "package": pkg
-                            }
-                    else:
-                        versions = self.CVE_DATA['pypi'][pkg]['all_ver']
-                    if versions:
-                        logger.info("Processing {}".format(data['id']))
-                        data['ecosystem'] = "pypi"
-                        pypi_pkg_versions[pkg] = versions
-                        vuln_versions = self._get_semver_versions(data['vulnerableVersions'])
-                        data['rules'] = self._get_version_rules(vuln_versions)
-                        data['affected'] = self._get_affected_versions(data['rules'], versions)
-                        if len(data['affected']) == 0:
-                            logger.info("No active affected version found for {}. Ignored."
-                                        .format(pkg))
-                            continue
-                        total_vuln += 1
-                        del data['rules']
-                        self.CVE_DATA['pypi'][pkg]['affected'].extend(data['affected'])
-                        self.CVE_DATA['pypi'][pkg]['affected'] = list(
-                            set(self.CVE_DATA['pypi'][pkg]['affected']))
-                        premium = str(data['premium']).lower() == 'true'
-                        data['pvtVuln'] = premium
-                        data['description'] = re.sub("[\'\"]", "", data['description'])
-                        self.CVE_DATA['pypi'][pkg]['vulnerabilities'].append(data)
-
+                        self.CVE_DATA[eco][pkg]['vulnerabilities'].append(data)
                 except ValueError:
```

Comment:
What is expected behaviour here? Are we expected to continue after we log the exception?

Reply:
yes ideal way is to log and move forward